desc/compute/_basis_vectors.py

```diff
@@ -27,668 +27,3396 @@ def _e_sub_rho(params, transforms, profiles, data, **kwargs):
+            - 3 * data["R_r"] * (1 + data["omega_z"]) * data["omega_zz"]
+            - 3
+            * data["R"]
+            * (
+                data["omega_rz"] * data["omega_zz"]
+                + (1 + data["omega_z"]) * data["omega_rzz"]
+            )
+            + data["R_rzzz"]
+            - data["omega_r"]
+            * (
+                3 * (1 + data["omega_z"]) * data["R_zz"]
+                + 3 * data["R_z"] * data["omega_zz"]
+                - data["R"]
+                * (
+                    1
+                    + 3 * data["omega_z"]
+                    + 3 * data["omega_z"] ** 2
+                    + data["omega_z"] ** 3
+                    - data["omega_zzz"]
+                )
+            ),
+            3 * data["omega_rz"] * data["R_zz"]
+            + 3 * (1 + data["omega_z"]) * data["R_rzz"]
+            + 3 * data["R_rz"] * data["omega_zz"]
+            + 3 * data["R_z"] * data["omega_rzz"]
+            - data["R_r"]
+            * (
+                1
+                + 3 * data["omega_z"]
+                + 3 * data["omega_z"] ** 2
+                + data["omega_z"] ** 3
+                - data["omega_zzz"]
+            )
+            - data["R"]
+            * (
+                3 * data["omega_rz"] * (1 + data["omega_z"] * (1 + data["omega_z"]))
+                - data["omega_rzzz"]
+            )
+            + data["omega_r"]
+            * (
+                -3 * data["R_z"] * (1 + data["omega_z"]) ** 2
+                - 3 * data["R"] * (1 + data["omega_z"]) * data["omega_zz"]
+                + data["R_zzz"]
+            ),
+            data["Z_rzzz"],
+        ]
+    ).T
     return data
 
 
 @register_compute_fun(
-    name="e_zeta_rz",
-    label="\\partial_{\\rho \\zeta} \\mathbf{e}_{\\zeta}",
-    units="m",
-    units_long="meters",
-    description="Covariant toroidal basis vector, second derivative wrt radial "
-    + "coordinate and toroidal angle",
+    name="e^rho_r",
+    label="\\partial{\\rho} \\mathbf{e}^{\\rho}",
+    units="m^{-1}",
+    units_long="inverse meters",
+    description="Contravariant radial basis vector, derivative wrt radial coordinate",
     dim=3,
     params=[],
     transforms={},
     profiles=[],
     coordinates="rtz",
-    data=["R_rz", "R_rzz", "Z_rzz"],
+    data=["e_theta", "e_zeta", "e_theta_r", "e_zeta_r", "sqrt(g)", "sqrt(g)_r"],
+    axis_limit_data=["e_theta_rr", "sqrt(g)_rr"],
 )
-def _e_sub_zeta_rz(params, transforms, profiles, data, **kwargs):
-    data["e_zeta_rz"] = jnp.array([data["R_rzz"], data["R_rz"], data["Z_rzz"]]).T
+def _e_sup_rho_r(params, transforms, profiles, data, **kwargs):
+    a = cross(data["e_theta_r"], data["e_zeta"])
+    data["e^rho_r"] = transforms["grid"].replace_at_axis(
+        (
+            (a + cross(data["e_theta"], data["e_zeta_r"])).T / data["sqrt(g)"]
+            - cross(data["e_theta"], data["e_zeta"]).T
+            * data["sqrt(g)_r"]
+            / data["sqrt(g)"] ** 2
+        ).T,
+        lambda: (
```

Comment:
Ok I'll add the comment for now. It's necessary to defer them because if the grid doesn't have an point at the axis those keys aren't in the dictionary because those dependencies were not computed.